### DISCUSSION TITLE
Applying hl7 message filter to one destination, the filter is applied to the next destination

### DISCUSSION THREAD

**Discussion by HassanBosha** [nextgenhealthcare/connect#6165] [Q&A] [answered]
Upon Applying a filter to a destination that is in between many destination the messages are altered (the messages come from source to the destination directly), but when it comes from source to a destination that has a filter, the next destination receives an altered message (filtered from the above destination.

**Comment by pacmano1** [comment]
I highly doubt that.

Provide a channel export where you see this behavior.

**Reply by HassanBosha** [reply]
[Test Channel.zip](https://github.com/nextgenhealthcare/connect/files/14997714/Test.Channel.zip)

**Reply by pacmano1** [reply, marked as answer]
You are storing messages to $c, **you don't have to do that at all.**    Since you are not transforming msg on the source side, each destiantion can just process "msg".

It looks like you just want to send message to different destinations based on their type, your code should be far more simple.

What is likely happening here is you have queueing enabled and on failure your $c var being converted in the transformer isn't convertable in the same way to HL7 anymore based on how it is stored in the database waiting to be resent.

Remove all queuing on all destinations and see if this happens stil. 

Please read the documetattion on ```destinationSet.``` commands which is better in your use case.

```JS
$c('message_code',msg['MSH']['MSH.9']['MSH.9.1'].toString())

if ($c('message_code') == 'ORM') {
	destinationSet.removeAllExcept(['ORM to A', 'ORM to B']);  // these are the name of the destinations that send ORMs
}

if ($c('message_code') == 'ORU') {
	destinationSet.removeAllExcept(['ORU to A', 'ORU to B']);  // these are the name of the destinations that send ORUs
}
```
Your destinatiosn would have `${message.encodedData}` in the template.
